src/axom/quest/MarchingCubes.cpp

```diff
@@ -0,0 +1,895 @@
+        m_coordsViews[d] =
+          axom::ArrayView<const double, DIM>(coordsPtr, m_pShape, coordSp);
+      }
+    }
+
+    // Nodal function
+    {
+      auto& fcnValues = m_dom->fetch_existing(fcnPath + "/values");
+      const double* fcnPtr = fcnValues.as_double_ptr();
+      m_fcnView = axom::ArrayView<const double, DIM>(fcnPtr, m_pShape);
+    }
+
+    // Mask
+    {
+      const int* maskPtr = nullptr;
+      if(!maskPath.empty())
+      {
+        auto& maskValues = m_dom->fetch_existing(maskPath + "/values");
+        maskPtr = maskValues.as_int_ptr();
+      }
+      if(maskPtr)
+      {
+        m_maskView = axom::ArrayView<const int, DIM>(maskPtr, m_cShape);
+      }
+    }
+
+    m_caseIds =
+      axom::Array<std::uint16_t, DIM, axom::MemorySpace::Dynamic>(m_cShape);
+  }
+
+  /*!
+    @brief Implementation of virtual mark_crossings.
+
+    Virtual methods cannot be templated, so this implementation
+    delegates to methods templated on DIM.
+  */
+  void mark_crossings() override { mark_crossings_dim(); }
+
+// RAJA_INDEX_VALUE_T doesn't work on rzansel.  Maybe old RAJA?
+#define USE_RAJA_INDEX_VALUE_TYPES 0
+
+  //!@brief Populate m_caseIds with crossing indices.
+  template <int TDIM = DIM>
+  typename std::enable_if<TDIM == 2>::type mark_crossings_dim()
+  {
+#if USE_RAJA_INDEX_VALUE_TYPES == 1
+    RAJA_INDEX_VALUE_T(IIDX, axom::IndexType, "IIDX");
+    RAJA_INDEX_VALUE_T(JIDX, axom::IndexType, "JIDX");
+    RAJA::TypedRangeSegment<JIDX> jRange(0, m_cShape[0]);
+    RAJA::TypedRangeSegment<IIDX> iRange(0, m_cShape[1]);
+#else
+    RAJA::RangeSegment jRange(0, m_cShape[0]);
+    RAJA::RangeSegment iRange(0, m_cShape[1]);
+#endif
+    using EXEC_POL = RAJA::KernelPolicy<
+      // TODO: Why does LoopPolicy here not compile on rzansel with cuda?  I have to use RAJA::seq_exec
+      RAJA::statement::
+        For<1, LoopPolicy, RAJA::statement::For<0, LoopPolicy, RAJA::statement::Lambda<0>>>>;
+    RAJA::kernel<EXEC_POL>(
+      RAJA::make_tuple(iRange, jRange),
+#if USE_RAJA_INDEX_VALUE_TYPES == 1
+      // TODO: Why does AXOM_LAMBDA here not compile on rzansel with cuda?
+      AXOM_LAMBDA(IIDX it, JIDX jt) {
+        auto& i = *it;
+        auto& j = *jt;
+#else
+      AXOM_LAMBDA(int i, int j) {
+#endif
+        const bool skipZone = !m_maskView.empty() && bool(m_maskView(j, i));
+        if(!skipZone)
+        {
+          // clang-format off
+          double nodalValues[CELL_CORNER_COUNT] =
+            {m_fcnView(j    , i    ),
+             m_fcnView(j    , i + 1),
+             m_fcnView(j + 1, i + 1),
+             m_fcnView(j + 1, i    )};
+          // clang-format on
+
+          auto crossingCase = compute_crossing_case(nodalValues);
+          m_caseIds(j, i) = crossingCase;
+          // std::cout<<"cell ("<<i<<','<<j<<") "<<(&m_fcnView(j,i)-&m_fcnView(0,0))<<"  "<<m_fcnView(j,i)<<std::endl; // confirm that the indices are correct and I am accessing memory with stride-1.
```

Comment:
Unless this comment is valuable for future debugging, please remove (along with the one at line 244).